src/process.cr

```diff
@@ -415,6 +415,70 @@ class Process
   private def close_io(io)
     io.close if io
   end
+
+  # Sets the real, effective, and saved user to the one specified.
+  def self.become(user : User) : Nil
+    return if (LibC.setuid(user.uid) == 0)
+    Errno.new("The calling process was not privileged.")
+  end
+
+  # Sets the real, effective, and saved group to the one specified.
+  def self.become(group : Group) : Nil
+    return if (LibC.setgid(group.gid) == 0)
+    Errno.new("The calling process was not privileged.")
+  end
+
+  # Sets the real, effective, and saved user and group to the ones specified.
+  def self.become(user : User, group : Group) : Nil
+    become(group)
+    become(user)
+  end
+
+  # Returns the current process's user.
+  def self.user : User
+    return User[LibC.getuid]
+  end
+
+  # Returns the current process's group.
+  def self.group : Group
+    return Group[LibC.getgid]
+  end
+
+  # Returns the current process's effective user.
+  def self.effective_user : User
+    return User[LibC.geteuid]
+  end
+
+  # Returns the current process's effective group.
+  def self.effective_group : Group
+    return Group[LibC.getegid]
+  end
+
+  # Returns the current process's saved user.
+  #
+  # Note: Not all Unix and Unix deriviatives support a saved user.
+  def self.saved_user : User
+    {% if flag?(:openbsd) || flag?(:freebsd) || flag?(:linux) %}
+      suid = uninitialized LibC::UidT
+      return User[suid] if ( LibC.getresuid(out ruid, out euid, pointerof(suid)) == 0 )
+      raise Errno.new("Failed to get saved user")
+    {% else %}
+      raise NotImplementedError.new("Process.saved_user")
```

Comment:
IMHO a non implemented method should raise at compile time since the program won't work.

Reply:
We've discussed this in another issue and raising at runtime is consistent with what we currently do in Crystal::System.

Reply:
See: https://github.com/crystal-lang/crystal/issues/4905